CHANGELOG.md

```diff
@@ -1,29 +1,29 @@
-## 0.4.0
+## 0.4.0 - 2023-04-13
```

Comment:
Adding these because `cider` requires a specific `CHANGELOG` format. At this point I don't think we're using `cider describe <version>` to output release notes which requires parsing the `CHANGELOG` (we're only using `cider` to set and read the version from `pubspec.yaml`) but this makes it so that we can.